stdlib/public/core/AutoDiff.swift

```diff
@@ -34,24 +42,56 @@ public protocol VectorProtocol : AdditiveArithmetic {
 }
 
 public extension VectorProtocol {
+  mutating func add(_ x: VectorSpaceScalar) -> Self {
+    self = adding(x)
+  }
+
+  static func + (lhs: Self, rhs: VectorSpaceScalar) -> Self {
+    lhs.adding(rhs)
+  }
+
+  static func + (lhs: VectorSpaceScalar, rhs: Self) -> Self {
+    rhs.adding(lhs)
+  }
+
+  static func += (lhs: inout Self, rhs: VectorSpaceScalar) {
+    lhs.add(rhs)
+  }
+
+  mutating func subtract(_ x: VectorSpaceScalar) -> Self {
+    self = subtracting(x)
+  }
+
+  static func - (lhs: Self, rhs: VectorSpaceScalar) -> Self {
+    lhs.subtracting(rhs)
+  }
+
+  static func -= (lhs: inout Self, rhs: VectorSpaceScalar) {
+    lhs.subtract(rhs)
+  }
+
   mutating func scale(by scalar: VectorSpaceScalar) {
     self = scaled(by: scalar)
   }
 
   static func * (lhs: Self, rhs: VectorSpaceScalar) -> Self {
-    return lhs.scaled(by: rhs)
+    lhs.scaled(by: rhs)
   }
 
   static func * (lhs: VectorSpaceScalar, rhs: Self) -> Self {
-    return rhs.scaled(by: lhs)
+    rhs.scaled(by: lhs)
   }
 
   static func *= (lhs: inout Self, rhs: VectorSpaceScalar) {
     lhs.scale(by: rhs)
   }
 }
 
-public extension VectorProtocol where VectorSpaceScalar: SignedNumeric {
+public extension VectorProtocol where VectorSpaceScalar : SignedNumeric {
+  static func - (lhs: VectorSpaceScalar, rhs: Self) -> Self {
+    -rhs.adding(lhs)
```

Comment:
I think this would be cleaner 
```suggestion
    lhs.subtracting(rhs)
```